src/realm/object-store/impl/realm_coordinator.cpp

```diff
@@ -278,7 +278,7 @@ std::shared_ptr<Realm> RealmCoordinator::get_realm(Realm::Config config, util::O
     return realm;
 }
 
-std::shared_ptr<Realm> RealmCoordinator::get_realm(std::shared_ptr<util::Scheduler> scheduler)
+std::shared_ptr<Realm> RealmCoordinator::get_realm(std::shared_ptr<util::Scheduler> scheduler, bool first_time_open)
```

Comment:
In most cases, it is an anti-pattern to tack on a bool parameter to a method, can this be designed in different way to reduce the clutter? How about putting it in the config?

Reply:
This value is coming from the async task (which resulted when open_db was called in get_synchronized_realm), so it's not really possible.

Reply:
We could add another bool to `SyncConfig` and set that inside `RealmCoordinator::get_synchronized_realm` but that is ugly too...

Reply:
This isn't something that makes sense as a configuration field. Whether or not this happens to be the first Realm opened from the config in this run of the process isn't a configuration setting.

Reply:
OK, let's leave this as it is. The idea of passing this boolean is not the best, another possibility could be to have a state variable inside the coordinator, in order to track whether the file was opened for the first time or not. 
But also this does not seem great.